Address safer CPP warnings in WKDownload

cdumez opened the pull request:
#### 443f5cba747191382d00c39c2aa8fc36081529cd
<pre>
Address safer CPP warnings in WKDownload
<a href="https://bugs.webkit.org/show_bug.cgi?id=300598">https://bugs.webkit.org/show_bug.cgi?id=300598</a>

Reviewed by Darin Adler.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/Shared/Cocoa/WKObject.h:
* Source/WebKit/UIProcess/API/Cocoa/WKDownload.mm:
(-[WKDownload protectedDownload]):
(-[WKDownload cancel:]):
(-[WKDownload webView]):
(-[WKDownload setDelegate:]):
(-[WKDownload progress]):
(-[WKDownload dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKDownload.mm:
(-[_WKDownload initWithDownload2:]):
(-[_WKDownload cancel]):
(-[_WKDownload publishProgressAtURL:]):
(-[_WKDownload originatingWebView]):
* Source/WebKit/UIProcess/API/Cocoa/_WKDownloadInternal.h:
* Source/WebKit/UIProcess/Downloads/DownloadProxy.h:

Canonical link: <a href="https://commits.webkit.org/301443@main">https://commits.webkit.org/301443@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c17b20a472a4c8b36dc2e4fc9c97e27c0a210d89

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/125880 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45542 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/36296 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/132750 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/77748 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/76330310-4c78-4ef5-ac7a-f953b55e0f37) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/46225 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/54100 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95899 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/64003 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/7c66a7ba-14b8-4968-bee9-f2becc5086ac) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/128828 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36961 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/112558 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/76390 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/d69f62b5-1ba7-4ccc-ba91-bb44e81b9cd2) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35869 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/76222 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/106742 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30955 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/135432 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/52665 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/40398 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/104365 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/53114 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/108768 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104092 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26533 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/49465 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/27779 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/50032 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/52559 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/58371 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51909 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/55254 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/53602 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebKit/UIProcess/API/Cocoa/WKDownload.mm:
```diff
@@ -64,11 +64,12 @@ explicit DownloadClient(id<WKDownloadDelegatePrivate> delegate)
 
         RetainPtr<NSURLRequest> nsRequest = request.nsURLRequest(WebCore::HTTPBodyUpdatePolicy::DoNotUpdateHTTPBody);
-        [m_delegate download:wrapper(download) willPerformHTTPRedirection:checked_objc_cast<NSHTTPURLResponse>(response.nsURLResponse()) newRequest:nsRequest.get() decisionHandler:makeBlockPtr([request = WTFMove(request), completionHandler = WTFMove(completionHandler), checker = WebKit::CompletionHandlerCallChecker::create(m_delegate.get().get(), @selector(download:willPerformHTTPRedirection:newR ...
+        [delegate download:protectedWrapper(download).get() willPerformHTTPRedirection:RetainPtr { checked_objc_cast<NSHTTPURLResponse>(response.nsURLResponse()) }.get() newRequest:nsRequest.get() decisionHandler:makeBlockPtr([request = WTFMove(request), completionHandler = WTFMove(completionHandler), checker = WebKit::CompletionHandlerCallChecker::create(m_delegate.get().get(), @selector(downloa ...
```
Might want to use the `retainPtr` function template in cases like this. not sure which is better.
```suggestion
        [delegate download:protectedWrapper(download).get() willPerformHTTPRedirection:retainPtr(checked_objc_cast<NSHTTPURLResponse>(response.nsURLResponse())).get() newRequest:nsRequest.get() decisionHandler:makeBlockPtr([request = WTFMove(request), completionHandler = WTFMove(completionHandler), checker = WebKit::CompletionHandlerCallChecker::create(m_delegate.get().get(), @selector(download:willPerformHTTPRedirection:newRequest:decisionHandler:))](WKDownloadRedirectPolicy policy) mutable {
```

cdumez replied:
Really? I was actually thinking we should get rid of the `retainPtr()` function as it doesn't seem useful now that we don't need to specify the template parameter for `RetainPtr`. I don't think the function brings much value anymore personally.

darinadler commented on Source/WebKit/UIProcess/API/Cocoa/WKDownload.mm:
```diff
@@ -183,39 +186,43 @@ void decidePlaceholderPolicy(WebKit::DownloadProxy& download, CompletionHandler<
-        NSProgress *progress = wrapper(download).progress;
-        progress.totalUnitCount = totalBytesExpectedToWrite;
-        progress.completedUnitCount = totalBytesWritten;
+        RetainPtr<NSProgress> progress = protectedWrapper(download).get().progress;
```
Can we leave out the class name?
```suggestion
        RetainPtr progress = protectedWrapper(download).get().progress;
```

cdumez replied:
No, we cannot. My first iteration actually had `NSProgress` omitted and I had to add it back so it would compile.

cdumez committed:
Address safer CPP warnings in WKDownload
https://bugs.webkit.org/show_bug.cgi?id=300598

Reviewed by Darin Adler.

* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UnretainedCallArgsCheckerExpectations:
* Source/WebKit/Shared/Cocoa/WKObject.h:
* Source/WebKit/UIProcess/API/Cocoa/WKDownload.mm:
(-[WKDownload protectedDownload]):
(-[WKDownload cancel:]):
(-[WKDownload webView]):
(-[WKDownload setDelegate:]):
(-[WKDownload progress]):
(-[WKDownload dealloc]):
* Source/WebKit/UIProcess/API/Cocoa/_WKDownload.mm:
(-[_WKDownload initWithDownload2:]):
(-[_WKDownload cancel]):
(-[_WKDownload publishProgressAtURL:]):
(-[_WKDownload originatingWebView]):
* Source/WebKit/UIProcess/API/Cocoa/_WKDownloadInternal.h:
* Source/WebKit/UIProcess/Downloads/DownloadProxy.h:

Canonical link: https://commits.webkit.org/301443@main
